src/main/scala/goatrodeo/util/ArtifactWrapper.scala

```diff
@@ -37,26 +71,42 @@ case class FileWrapper(f: File, deleteOnFinalize: Boolean)
 
   override def isFile(): Boolean = f.isFile()
 
-  override def listFiles(): Vector[ArtifactWrapper] =
-    f.listFiles().toVector.filter(!_.getName().startsWith(".")).map(FileWrapper(_, false))
+  override def listFiles(): Vector[ArtifactWrapper] = {
+    // the javadocs for `listFiles` says that if it is not a directory, it returns `null`… we need to guard against that
+    Option(f.listFiles()) match {
+      case Some(files) =>
+        files.toVector
+          .filter(!_.getName().startsWith("."))
```

Comment:
Do you know why we filter out dotfiles here?

Reply:
Turns out that Coursier and other artifact retrieval tools keep a lot of meta information in `.` files. In order to get some stuff working during the great indexing of Maven Central, I had to filter out the `.` files.

Thinking about it... filtering them out for real filesystem directories ("put all your files in directories and subdirectories and run Goat Rodeo") makes sense. Filtering out `.` files and directories in artifacts themselves (an ISO is an artifact, a tar is an artifact) does not make sense.

Enhancing the code based on the above would be a win